### PR TITLE
DependencyOrder: Fix var initialization bug

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
https://github.com/Homebrew/brew/issues/4395
cc @ilovezfs

### CODE REVIEW

File: Library/Homebrew/rubocops/dependency_order_cop.rb [hunk 1]
```diff
@@ -58,9 +58,8 @@ def sort_conditional_dependencies!(ordered)
             ordered.each_with_index do |dep, pos|
               idx = pos+1
               match_nodes = build_with_dependency_name(dep)
-              idx1 = pos if match_nodes && !match_nodes.empty?
```
Comment (GauthamGoli) [inline Library/Homebrew/rubocops/dependency_order_cop.rb]: `idx1` was supposed to be initialized as `nil` for each iteration, but due to this buggy code, it had value from previous iteration and that was causing the bug.

**Comment by MikeMcQuaid** [comment]
Thanks @GauthamGoli! Future note: please create PRs from your branch rather than a fork. Ta 🙇

**Comment by MikeMcQuaid** [comment]
@GauthamGoli and also if you use "Fixes #4395" then it'll autoclose on merge 😎

**Comment by GauthamGoli** [comment]
@MikeMcQuaid Oh my bad. I specified wrong remote when pushing the branch 🙈Apologies.

**Pull request merged**